packages/metadata/src/Decorated/storage/fromMetadata/createFunction.ts

```diff
@@ -157,7 +157,7 @@ function extendHeadMeta (registry: Registry, { meta: { documentation, name, type
     type: registry.createType('StorageEntryTypeLatest', registry.createType('PlainTypeLatest', type.isMap ? type.asMap.key : type.asDoubleMap.key1), 0)
   });
 
-  return registry.createType('StorageKey', iterFn, { method, section });
+  return (arg?: any) => registry.createType('StorageKey', iterFn(arg), { method, section });
```

Comment:
One small issue I see here is  when no arguments,`StorageKey` will need to be recreated. Previously it was cached.

```
const prefixKey = registry.createType('StorageKey', iterFn(), { method, section });
return (arg?: any) => !isUndefined(arg) && !isNull(arg)
  ? registry.createType('StorageKey', iterFn(arg), { method, section })
  : prefixKey;
```